esmvaltool/cmorizers/data/formatters/datasets/cds_uerra.py

```diff
@@ -234,10 +231,15 @@ def _regrid_dataset(in_dir, var, cfg):
         assert int((input_da == 0.).sum()) == 0  # Make sure that there
         # are no zero's in the data,
         # since they will be masked out
-        regridder = xe.Regridder(input_ds,
-                                 targetgrid_ds,
-                                 'bilinear',
-                                 reuse_weights=True)
+        shapes = (input_ds["lon"].shape, input_ds["lat"].shape,
+                  targetgrid_ds["lon"].shape, targetgrid_ds["lat"].shape)
+        if regridder is None:
+            regridder = xe.Regridder(input_ds,
+                                     targetgrid_ds,
+                                     'bilinear')
+            ref_shapes = shapes
+        else:
+            assert shapes == ref_shapes
```

Comment:
am slightly confused by the way `regridder` is used as a variable - you are setting it to `None`, so it'll always be `none` - why the if/else block?

Reply:
Ah, but it's in a loop. It will only be `None` in the first round of the loop. Then, the regridder will be reused for all other files.